### PR TITLE
ZH-2569: add nctl upgrade test 9

### LINKED ISSUE #2569: Add nctl test for joining node with trusted hash from before the upgrade.
Split from #2476.

Scenario:

- 1.5.x (fast-sync) network is running
- remember the trusted hash
- upgrade network to 1.6.x (next version of the fast-sync node)
- try to join a new node with trusted hash from 1.5
- the node is expected to fail with a message

### PR DESCRIPTION
**Pull request by TomVasile**
Changes:
- Adds new nctl upgrade test 9
- Adds test 9 to nightly test runs

Ticket: https://app.zenhub.com/workspaces/engineering-60953fafb1945f0011a3592d/issues/casper-network/casper-node/2569

### CODE REVIEW

File: utils/nctl/sh/scenarios-upgrades/upgrade_scenario_09.sh [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+function _step_08()
+{
+    local NODE_ID=${1}
+    local NODE_PATH
+
+    log_step_upgrades 8 "Checking for failed node-$NODE_ID..."
+
+    NODE_PATH="$(get_path_to_node $NODE_ID)"
+
+    if ( cat "$NODE_PATH"/logs/stdout.log | grep -q 'the trusted block has an older version' ); then
```
Comment (rafal-ch) [inline utils/nctl/sh/scenarios-upgrades/upgrade_scenario_09.sh]: We repeat the string `the trusted block has an older version` three times here. Please consider replacing it with a variable.

**Review by rafal-ch — APPROVED** [review]
LGTM :+1: with a small nit

File: utils/nctl/sh/scenarios-upgrades/upgrade_scenario_09.sh (hunk 1 above)
Comment (TomVasile) [reply utils/nctl/sh/scenarios-upgrades/upgrade_scenario_09.sh]: addressed in: https://github.com/casper-network/casper-node/pull/2693/commits/88678e4a3b00bfb124ad3ff9177da924e0677242

**Pull request merged**